types/lib/legacy/class.d.ts

```diff
@@ -0,0 +1,110 @@
+   * features (key: property names, value: object containing property metadata).
+   * Valid keys in per-property metadata include:
+   * - `type` (String|Number|Object|Array|...): Used by
+   *   `attributeChangedCallback` to determine how string-based attributes
+   *   are deserialized to JavaScript property values.
+   * - `notify` (boolean): Causes a change in the property to fire a
+   *   non-bubbling event called `<property>-changed`. Elements that have
+   *   enabled two-way binding to the property use this event to observe changes.
+   * - `readOnly` (boolean): Creates a getter for the property, but no setter.
+   *   To set a read-only property, use the private setter method
+   *   `_setProperty(property, value)`.
+   * - `observer` (string): Observer method name that will be called when
+   *   the property changes. The arguments of the method are
+   *   `(value, previousValue)`.
+   * - `computed` (string): String describing method and dependent properties
+   *   for computing the value of this property (e.g. `'computeFoo(bar, zot)'`).
+   *   Computed properties are read-only by default and can only be changed
+   *   via the return value of the computing method.
+   *
+   * `observers`: Array of strings describing multi-property observer methods
+   *  and their dependent properties (e.g. `'observeABC(a, b, c)'`).
+   *
+   * `listeners`: Object describing event listeners to be added to each
+   *  instance of this element (key: event name, value: method name).
+   *
+   * `behaviors`: Array of additional `info` objects containing metadata
+   * and callbacks in the same format as the `info` object here which are
+   * merged into this element.
+   *
+   * `hostAttributes`: Object listing attributes to be applied to the host
+   *  once created (key: attribute name, value: attribute value).  Values
+   *  are serialized based on the type of the value.  Host attributes should
+   *  generally be limited to attributes such as `tabIndex` and `aria-...`.
+   *  Attributes in `hostAttributes` are only applied if a user-supplied
+   *  attribute is not already present (attributes in markup override
+   *  `hostAttributes`).
+   *
+   * In addition, the following Polymer-specific callbacks may be provided:
+   * - `registered`: called after first instance of this element,
+   * - `created`: called during `constructor`
+   * - `attached`: called during `connectedCallback`
+   * - `detached`: called during `disconnectedCallback`
+   * - `ready`: called before first `attached`, after all properties of
+   *   this element have been propagated to its template and all observers
+   *   have run
+   *
+   * @returns Generated class
+   */
+  function Class(info: PolymerInit): {new(): HTMLElement};
+}
+
+declare class PolymerGenerated {
+  created(): any;
+  _registered(): any;
+  _applyListeners(): any;
+
+  /**
+   * only apply if not already set.
+   */
+  _ensureAttributes(): any;
```

Comment:
I think all of these should return `void` too

Reply:
Fixed in https://github.com/Polymer/polymer/pull/4998